dyno-queues-redis/src/main/java/com/netflix/dyno/queues/redis/RedisDynoQueue.java

```diff
@@ -593,12 +591,13 @@ private String getUnackKey(String queueName, String shard) {
 
             if (e.getCause() instanceof DynoException) {
                 if (retryCount < this.retryCount) {
-                    return executeWithRetry(r, ++retryCount);
+                    return executeWithRetry(opName, keyName, r, ++retryCount);
                 }
             }
             throw new RuntimeException(e.getCause());
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            throw new RuntimeException(
+                    "Operation: ( " + opName + " ) failed on key: [" + keyName + " ]. Cause:\n" + e.getCause());
```

Comment:
Exception cause can be stacked on RunTimeException with a different construction. https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html#RuntimeException(java.lang.String,%20java.lang.Throwable). You can let the caller decide to print the stack trace.

Reply:
Done.